shopinvader_search_engine_product_stock/models/product_product.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 Akretion (http://www.akretion.com)
+# Copyright 2018 ACSONE SA/NV
+# Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from collections import defaultdict
+
+from odoo import api, models
+
+from odoo.addons.base_sparse_field.models.fields import Serialized
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    stock_data = Serialized(compute="_compute_stock_data")
```

Comment:
```suggestion
    stock_data = fields.Json(compute="_compute_stock_data")
```
same functionality for computed fields but without additional dependency on base_sparse_field.